spec/policy/rate_limit/rate_limit_spec.lua

```diff
@@ -27,6 +27,27 @@ local function is_gt(_, arguments)
 end
 assert:register("matcher", "gt", is_gt)
 
+
+local s = require('say')
+local util = require ('luassert.util')
+
+local function returns_error(state, arguments, level)
+  local argcnt = arguments.n
+  assert(argcnt == 3, s("assertion.internal.argtolittle", { "error_matches", 3, tostring(argcnt) }), level)
+
+  local expected = tonumber(arguments[1])
+  local ok = tonumber(arguments[2])
+  local actual = tonumber(arguments[3])
+
+  local result = not ok and expected == actual
+  -- switch arguments for proper output message
+  util.tinsert(arguments, 1, util.tremove(arguments, 3))
+  state.failure_message = arguments[3]
+  return result
+end
+
+assert:register("assertion", "returns_error", returns_error, "assertion.error.positive", "assertion.error.negative")
```

Comment:
`assertion.error.positive` and `assertion.error.negative` are not set.

Reply:
But it worked. I just copied it from luassert internals, so I assume they are defined there and just reused.

I plan to extract this to a helper, so it can be reused in other files too.